Demo/src/main/java/com/paypal/android/ui/vault/VaultFragment.kt

```diff
@@ -0,0 +1,264 @@
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.ViewCompositionStrategy
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.navArgs
+import com.paypal.android.api.model.PaymentToken
+import com.paypal.android.api.model.SetupToken
+import com.paypal.android.api.services.SDKSampleServerAPI
+import com.paypal.android.cardpayments.Card
+import com.paypal.android.cardpayments.CardClient
+import com.paypal.android.cardpayments.VaultListener
+import com.paypal.android.cardpayments.VaultRequest
+import com.paypal.android.cardpayments.VaultResult
+import com.paypal.android.corepayments.CoreConfig
+import com.paypal.android.corepayments.PayPalSDKError
+import com.paypal.android.ui.card.DateString
+import com.paypal.android.uishared.components.PaymentTokenView
+import com.paypal.android.uishared.components.PropertyView
+import com.paypal.android.uishared.components.SetupTokenView
+import com.paypal.android.usecase.CreatePaymentTokenUseCase
+import com.paypal.android.usecase.CreateSetupTokenUseCase
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class VaultFragment : Fragment() {
+
+    @Inject
+    lateinit var createPaymentTokenUseCase: CreatePaymentTokenUseCase
+
+    @Inject
+    lateinit var createSetupTokenUseCase: CreateSetupTokenUseCase
+
+    @Inject
+    lateinit var sdkSampleServerAPI: SDKSampleServerAPI
+
+    private lateinit var cardClient: CardClient
+
+    private val args: VaultFragmentArgs by navArgs()
+    private val viewModel by viewModels<VaultViewModel>()
+
+    @ExperimentalMaterial3Api
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        args.prefillCard?.card?.let { viewModel.prefillCard(it) }
+
+        return ComposeView(requireContext()).apply {
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                MaterialTheme {
+                    Surface(modifier = Modifier.fillMaxSize()) {
+                        val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+                        VaultView(
+                            uiState = uiState,
+                            onCreateSetupTokenSubmit = { createSetupToken() },
+                            onAttachCardToSetupTokenSubmit = { attachCardToSetupToken() },
+                            onCreatePaymentTokenSubmit = { createPaymentToken() },
```

Comment:
Same Q as on [iOS](https://github.com/paypal/paypal-ios/pull/172/files#r1291431478) - do we need to display the paymentToken in our Demo? Also, i'm still unclear on what a merchant is supposed to do with a paymentToken on their server after they get it

Reply:
The demo app is designed to highlight interactions between client and server. Displaying the payment token is evidence that the vault completed successfully. A payment token can be used as a `payment_source` in PPaaS, I believe.